src/main/java/io/vertx/mqtt/impl/MqttConnection.java

```diff
@@ -0,0 +1,315 @@
+          break;
+
+        case PINGREQ:
+
+          this.handlePingreq();
+          break;
+
+        case DISCONNECT:
+
+          this.handleDisconnect();
+          break;
+
+        default:
+
+          this.channel.pipeline().fireExceptionCaught(new Exception("Wrong message type " + msg.getClass().getName()));
+          break;
+
+      }
+
+      // handling mapped Vert.x MQTT messages (from Netty ones) because they'll be provided
+      // to the higher layer (so need for ployglotization)
+    } else {
+
+      if (msg instanceof MqttSubscribeMessage) {
+
+        this.handleSubscribe((MqttSubscribeMessage) msg);
+
+      } else if (msg instanceof MqttUnsubscribeMessage) {
+
+        this.handleUnsubscribe((MqttUnsubscribeMessage) msg);
+
+      } else if (msg instanceof MqttPublishMessage) {
+
+        this.handlePublish((MqttPublishMessage) msg);
+
+      } else {
+
+        this.channel.pipeline().fireExceptionCaught(new Exception("Wrong message type"));
+      }
+    }
+  }
+
+  /**
+   * Used for calling the endpoint handler when a connection is established with a remote MQTT client
+   */
+  private void handleConnect(MqttConnectMessage msg) {
+
+    // retrieve will information from CONNECT message
+    MqttWillImpl will =
+      new MqttWillImpl(msg.variableHeader().isWillFlag(),
+        msg.payload().willTopic(),
+        msg.payload().willMessage(),
+        msg.variableHeader().willQos(),
+        msg.variableHeader().isWillRetain());
+
+    // retrieve authorization information from CONNECT message
+    MqttAuthImpl auth = (msg.variableHeader().hasUserName() &&
+      msg.variableHeader().hasPassword()) ?
+      new MqttAuthImpl(
+        msg.payload().userName(),
+        msg.payload().password()) : null;
+
+    // create the MQTT endpoint provided to the application handler
+    endpoint =
+      new MqttEndpointImpl(
+        this,
+        msg.payload().clientIdentifier(),
+        auth,
+        will,
+        msg.variableHeader().isCleanSession(),
+        msg.variableHeader().version(),
+        msg.variableHeader().name(),
+        msg.variableHeader().keepAliveTimeSeconds());
+
+    // keep alive == 0 means NO keep alive, no timeout to handle
+    if (msg.variableHeader().keepAliveTimeSeconds() != 0) {
+
+      // the server waits for one and a half times the keep alive time period (MQTT spec)
+      int timeout = msg.variableHeader().keepAliveTimeSeconds() +
+        msg.variableHeader().keepAliveTimeSeconds() / 2;
+
+      // modifying the channel pipeline for adding the idle state handler with previous timeout
+      channel.pipeline().addBefore("handler", "idle", new IdleStateHandler(0, 0, timeout));
+    }
+
+    endpointHandler.handle(endpoint);
+  }
+
+  /**
+   * Used for calling the subscribe handler when the remote MQTT client subscribes to topics
+   *
+   * @param msg message with subscribe information
+   */
+  synchronized void handleSubscribe(MqttSubscribeMessage msg) {
+
```

Comment:
@vietj my preference is that for method with more than one line, having a blank line helps makes it more readable. Removing the blank line is something related to the contribution rules for Vert.x ?